nexus/src/app/sagas/instance_create.rs

```diff
@@ -166,6 +192,76 @@ impl NexusSaga for SagaInstanceCreate {
             Ok(())
         }
 
+        for (i, group) in
+            params.create_params.affinity_groups.iter().enumerate()
+        {
+            let group = match group {
+                NameOrId::Id(id) => AffinityGroupUuid::from_untyped_uuid(*id),
+                _ => {
+                    return Err(SagaInitError::InvalidParameter(
+                        "Non-UUID group".to_string(),
+                    ));
+                }
+            };
```

Comment:
Yeah, if they're all IDs you don't even need this match.

Reply:
Agreed, this will require a new struct, and `InstanceCreate` already needs some clean-up. I can do this, but I'm probably going to save it for a follow-up PR?